qemu.go

```diff
@@ -217,32 +217,46 @@ func (q *qemu) appendFSDevices(devices []ciaoQemu.Device, podConfig PodConfig) [
 	return devices
 }
 
-func (q *qemu) appendConsoles(devices []ciaoQemu.Device, podConfig PodConfig) []ciaoQemu.Device {
+func (q *qemu) appendConsoles(devices []ciaoQemu.Device, podConfig *PodConfig) []ciaoQemu.Device {
 	serial := ciaoQemu.SerialDevice{
 		Driver: ciaoQemu.VirtioSerial,
 		ID:     "serial0",
 	}
 
 	devices = append(devices, serial)
 
+	if podConfig.Console == "" {
+		podConfig.Console = fmt.Sprintf("%s/%s/console.sock", runStoragePath, podConfig.ID)
+	}
+
+	console := ciaoQemu.CharDevice{
+		Driver:   ciaoQemu.Console,
+		Backend:  ciaoQemu.Socket,
+		DeviceID: "console0",
+		ID:       "charconsole0",
+		Path:     podConfig.Console,
+	}
+
+	devices = append(devices, console)
+
 	for i, c := range podConfig.Containers {
-		var console ciaoQemu.CharDevice
-		if c.Interactive == false || c.Console == "" {
-			consolePath := fmt.Sprintf("%s/%s/console.sock", runStoragePath, podConfig.ID)
+		// Need to add an offset because of the console created for the pod.
+		idx := i + 1
 
+		if c.Interactive == false || c.Console == "" {
 			console = ciaoQemu.CharDevice{
 				Driver:   ciaoQemu.Console,
 				Backend:  ciaoQemu.Socket,
-				DeviceID: fmt.Sprintf("console%d", i),
-				ID:       fmt.Sprintf("charconsole%d", i),
-				Path:     consolePath,
+				DeviceID: fmt.Sprintf("console%d", idx),
+				ID:       fmt.Sprintf("charconsole%d", idx),
+				Path:     fmt.Sprintf("%s/%s/%s/console.sock", runStoragePath, podConfig.ID, c.ID),
```

Comment:
@sboeuf is this the same as https://github.com/01org/cc-oci-runtime/blob/045d668b47ba23c8d6a2951c2fd8bdc9b8410f2c/src/oci.h#L66 
We use this for debugging the VM bootup and kernel issues too.